{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Clinical Disease Data EDA\n",
    "\n",
    "### Business Case:\n",
    "Your boss comes to you Monday morning and says “I figured out our next step; we are going to pivot from an online craft store and become a data center for genetic disease information! I found **ClinVar** which is a repository that contains expert curated data, and it is free for the taking. This is a gold mine! Look at the file and tell me what gene and mutation combinations are classified as dangerous.”\n",
    "\n",
    "Make sure that you only give your boss the dangerous mutations and include:\n",
    "\n",
    "1) Gene name\n",
    "\n",
    "2) Mutation ID number\n",
    "\n",
    "3) Mutation Position (chromosome & position)\n",
    "\n",
    "4) Mutation value (reference & alternate bases)\n",
    "\n",
    "5) Clinical significance (CLNSIG)\n",
    "\n",
    "6) Disease that is implicated\n",
    "\n",
    "As a final deliverable, you're planning to provide a dataframe with a short discussion of any specifics you plan to present to your boss (alongside the explanation of your results). You're also planning to limit your output to the first 100 harmful mutations and tell your boss how many total harmful mutations were found in the file.\n",
    "\n",
    "**Dataset: clinvar_final.txt file** (not disclosed here for privacy purposes)\n",
    "\n",
    "*Note: Missing values should be replaced with 'Not_Given'.*\n",
    "\n",
    "The unit of observation in this dataset is one row per mutation.\n",
    "\n",
    "A partial data dictionary for the dataset is available here: https://drive.google.com/file/d/1lx9yHdlcqmU_OlHiTUXKC_LQDqYBypH_/view?usp=sharing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### VCF File Description (Summarized from Version 4.1):\n",
    "\n",
    "```\n",
    "* The VCF specification:\n",
    "\n",
    "VCF is a text file format which contains meta-information lines, a header line, and then data lines each containing information about a position in the genome. The format also can contain genotype information on samples for each position.\n",
    "\n",
    "* Fixed fields:\n",
    "\n",
    "There are 8 fixed fields per record. All data lines are tab-delimited. In all cases, missing values are specified with a dot (‘.’). \n",
    "\n",
    "1. CHROM - chromosome number\n",
    "2. POS - position DNA nuceleotide count (bases) along the chromosome\n",
    "3. ID - The unique identifier for each mutation\n",
    "4. REF - reference base(s)\n",
    "5. ALT - alternate base(s)\n",
    "6. FILTER - filter status\n",
    "7. QUAL - quality\n",
    "8. INFO - a semicolon-separated series of keys with values in the format: <key>=<data>\n",
    "\n",
    "```\n",
    "### Applicable INFO Field Specifications:\n",
    "\n",
    "```\n",
    "GENEINFO = <Gene name>\n",
    "CLNSIG =  <Clinical significance>\n",
    "CLNDN = <Disease name>\n",
    "```\n",
    "\n",
    "### Sample ClinVar Data (VCF File Format):\n",
    "\n",
    "```\n",
    "##fileformat=VCFv4.1\n",
    "##fileDate=2019-03-19\n",
    "##source=ClinVar\n",
    "##reference=GRCh38\t\t\t\t\t\t\t\n",
    "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n",
    "1\t949523\trs786201005\tC\tT\t.\t.\tGENEINFO=ISG15;CLNSIG=5\n",
    "1\t949696\trs672601345\tC\tCG\t.\t.\tGENEINFO=ISG15;CLNSIG=5;CLNDN=Cancer\n",
    "1\t949739\trs672601312\tG\tT\t.\t.\tGENEINFO=ISG15;CLNDBN=Cancer\n",
    "1\t955597\trs115173026\tG\tT\t.\t.\tGENEINFO=AGRN;CLNSIG=2; CLNDN=Cancer\n",
    "1\t955619\trs201073369\tG\tC\t.\t.\tGENEINFO=AGG;CLNDN=Heart_dis \n",
    "1\t957640\trs6657048\tC\tT\t.\t.\tGENEINFO=AGG;CLNSIG=3;CLNDN=Heart_dis \n",
    "1\t976059\trs544749044\tC\tT\t.\t.\tGENEINFO=AGG;CLNSIG=0;CLNDN=Heart_dis \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014O42</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1O14122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>3</td>\n",
       "      <td>179210507</td>\n",
       "      <td>403908</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=393412;CLNDISDB=MedGen:C0018553,Orpha...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>3</td>\n",
       "      <td>179210511</td>\n",
       "      <td>526648</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=519163;CLNDISDB=MedGen:C0018553,Orpha...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>3</td>\n",
       "      <td>179210515</td>\n",
       "      <td>526640</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00002;ALLELEID=519178;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>3</td>\n",
       "      <td>179210516</td>\n",
       "      <td>246681</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=245287;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>3</td>\n",
       "      <td>179210538</td>\n",
       "      <td>259958</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>AF_EXAC=0.00001;ALLELEID=251013;CLNDISDB=MedGe...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS      ID REF ALT FILTER QUAL  \\\n",
       "0           1    1014O42  475283   G   A      .    .   \n",
       "1           1    1O14122  542074   C   T      .    .   \n",
       "2           1    1014143  183381   C   T      .    .   \n",
       "3           1    1014179  542075   C   T      .    .   \n",
       "4           1    1014217  475278   C   T      .    .   \n",
       "...       ...        ...     ...  ..  ..    ...  ...   \n",
       "102316      3  179210507  403908   A   G      .    .   \n",
       "102317      3  179210511  526648   T   C      .    .   \n",
       "102318      3  179210515  526640   A   C      .    .   \n",
       "102319      3  179210516  246681   A   G      .    .   \n",
       "102320      3  179210538  259958   A   T      .    .   \n",
       "\n",
       "                                                     INFO  \n",
       "0       AF_ESP=0.00546;AF_EXAC=0.00165;AF_TGP=0.00619;...  \n",
       "1       AF_ESP=0.00015;AF_EXAC=0.00010;ALLELEID=514926...  \n",
       "2       ALLELEID=181485;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "3       ALLELEID=514896;CLNDISDB=MedGen:C4015293,OMIM:...  \n",
       "4       AF_ESP=0.00515;AF_EXAC=0.00831;AF_TGP=0.00339;...  \n",
       "...                                                   ...  \n",
       "102316  ALLELEID=393412;CLNDISDB=MedGen:C0018553,Orpha...  \n",
       "102317  ALLELEID=519163;CLNDISDB=MedGen:C0018553,Orpha...  \n",
       "102318  AF_EXAC=0.00002;ALLELEID=519178;CLNDISDB=MedGe...  \n",
       "102319  AF_EXAC=0.00001;ALLELEID=245287;CLNDISDB=MedGe...  \n",
       "102320  AF_EXAC=0.00001;ALLELEID=251013;CLNDISDB=MedGe...  \n",
       "\n",
       "[102321 rows x 8 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 1: INITIAL SETUP - LIBRARY AND DATA FILE IMPORTS\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Import necessary libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import json\n",
    "\n",
    "# Set up pandas display options to display dataframes with 100 rows/columns maximum\n",
    "pd.set_option(\"display.max_rows\", 100, \"display.max_columns\", 100)\n",
    "pd.options.display.float_format = \"{:,.2f}\".format\n",
    "\n",
    "# Read vcf file (.txt format)\n",
    "# All data lines are tab-delimited, based on dataset documentation\n",
    "# Skip first 27 rows in dataset to eliminate meta-information lines\n",
    "clinical_df = pd.read_table(\"clinvar_final.txt\", delimiter = \"\\t\", skiprows = 27)\n",
    "\n",
    "# Display initial dataframe\n",
    "clinical_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 1 Notes/Observations:\n",
    "- vcf file/dataset imported and loaded correctly, with clean column names\n",
    "- Immediately apparent that this file is large based on its shape (102321 rows x 8 columns)\n",
    "- The INFO column appears to have several additional data fields embedded within it, which makes sense based on the documentation - will require some additional work to extract these fields\n",
    "- Looks like the data is a mix of numeric/integer and string data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe info:\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 102321 entries, 0 to 102320\n",
      "Data columns (total 8 columns):\n",
      " #   Column  Non-Null Count   Dtype \n",
      "---  ------  --------------   ----- \n",
      " 0   CHROM   102321 non-null  int64 \n",
      " 1   POS     102321 non-null  object\n",
      " 2   ID      102321 non-null  int64 \n",
      " 3   REF     102321 non-null  object\n",
      " 4   ALT     102321 non-null  object\n",
      " 5   FILTER  102321 non-null  object\n",
      " 6   QUAL    102321 non-null  object\n",
      " 7   INFO    102321 non-null  object\n",
      "dtypes: int64(2), object(6)\n",
      "memory usage: 6.2+ MB\n",
      "None\n",
      "\n",
      "Dataframe shape: (102321, 8)\n"
     ]
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 2a: CLEAN AND MERGE DATA AS NECESSARY TO PREP FOR EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Get basic info and Dtypes for dataframe, look for null values, get original shape to check dimensions, etc.\n",
    "print(\"Dataframe info:\")\n",
    "print(clinical_df.info(), end = \"\\n\\n\")\n",
    "print(\"Dataframe shape:\", clinical_df.shape)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 2a Notes/Observations:\n",
    "- No null values - makes sense and is expected given the documentation above, i.e., \"In all cases, missing values are specified with a dot (‘.’)\"\n",
    "- All expected columns are present based on the documentation above\n",
    "- CHROM column Dtype looks okay based on the documentation\n",
    "- POS column should have Dtype _int_ based on the documentation, but is currently Dtype _object_ - will need to investigate further\n",
    "- INFO column contains a large number of sub-components within it and is semicolon-delimited based on dataset documentation - will require an additional parsing step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNHGVS</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>RS</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.00546</td>\n",
       "      <td>0.00165</td>\n",
       "      <td>0.00619</td>\n",
       "      <td>446939</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014042G&gt;A</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>143888043</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.00015</td>\n",
       "      <td>0.00010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>514926</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014122C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>150861311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>181485</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014143C&gt;T</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1</td>\n",
       "      <td>786201005</td>\n",
       "      <td>OMIM_Allelic_Variant:147571.0003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>514896</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014179C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1553169766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.00515</td>\n",
       "      <td>0.00831</td>\n",
       "      <td>0.00339</td>\n",
       "      <td>446987</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014217C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>61766284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393412</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210507A&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1060500027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>519163</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210511T&gt;C</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1480813252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>519178</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome|Hereditary_cancer-predisposing...</td>\n",
       "      <td>NC_000003.12:g.179210515A&gt;C</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>199563773</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>245287</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210516A&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>753879573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>251013</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome|not_specified</td>\n",
       "      <td>NC_000003.12:g.179210538A&gt;T</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>201108344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         AF_ESP  AF_EXAC   AF_TGP ALLELEID  \\\n",
       "0       0.00546  0.00165  0.00619   446939   \n",
       "1       0.00015  0.00010      NaN   514926   \n",
       "2           NaN      NaN      NaN   181485   \n",
       "3           NaN      NaN      NaN   514896   \n",
       "4       0.00515  0.00831  0.00339   446987   \n",
       "...         ...      ...      ...      ...   \n",
       "102316      NaN      NaN      NaN   393412   \n",
       "102317      NaN      NaN      NaN   519163   \n",
       "102318      NaN  0.00002      NaN   519178   \n",
       "102319      NaN  0.00001      NaN   245287   \n",
       "102320      NaN  0.00001      NaN   251013   \n",
       "\n",
       "                                                 CLNDISDB  \\\n",
       "0        MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "1        MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "2        MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "3        MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "4        MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "...                                                   ...   \n",
       "102316  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102317  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102318  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102319  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102320  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "\n",
       "                                                    CLNDN  \\\n",
       "0       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "1       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "2       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "3       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "4       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "...                                                   ...   \n",
       "102316                                    Cowden_syndrome   \n",
       "102317                                    Cowden_syndrome   \n",
       "102318  Cowden_syndrome|Hereditary_cancer-predisposing...   \n",
       "102319                                    Cowden_syndrome   \n",
       "102320                      Cowden_syndrome|not_specified   \n",
       "\n",
       "                            CLNHGVS  \\\n",
       "0         NC_000001.11:g.1014042G>A   \n",
       "1         NC_000001.11:g.1014122C>T   \n",
       "2         NC_000001.11:g.1014143C>T   \n",
       "3         NC_000001.11:g.1014179C>T   \n",
       "4         NC_000001.11:g.1014217C>T   \n",
       "...                             ...   \n",
       "102316  NC_000003.12:g.179210507A>G   \n",
       "102317  NC_000003.12:g.179210511T>C   \n",
       "102318  NC_000003.12:g.179210515A>C   \n",
       "102319  NC_000003.12:g.179210516A>G   \n",
       "102320  NC_000003.12:g.179210538A>T   \n",
       "\n",
       "                                               CLNREVSTAT  \\\n",
       "0                     criteria_provided,_single_submitter   \n",
       "1                     criteria_provided,_single_submitter   \n",
       "2                          no_assertion_criteria_provided   \n",
       "3                     criteria_provided,_single_submitter   \n",
       "4                     criteria_provided,_single_submitter   \n",
       "...                                                   ...   \n",
       "102316                criteria_provided,_single_submitter   \n",
       "102317                criteria_provided,_single_submitter   \n",
       "102318  criteria_provided,_multiple_submitters,_no_con...   \n",
       "102319                criteria_provided,_single_submitter   \n",
       "102320  criteria_provided,_multiple_submitters,_no_con...   \n",
       "\n",
       "                        CLNSIG                      CLNVC     CLNVCSO  \\\n",
       "0                       Benign  single_nucleotide_variant  SO:0001483   \n",
       "1       Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "2                   Pathogenic  single_nucleotide_variant  SO:0001483   \n",
       "3       Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "4                       Benign  single_nucleotide_variant  SO:0001483   \n",
       "...                        ...                        ...         ...   \n",
       "102316  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102317           Likely_benign  single_nucleotide_variant  SO:0001483   \n",
       "102318  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102319  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102320           Likely_benign  single_nucleotide_variant  SO:0001483   \n",
       "\n",
       "           GENEINFO                             MC ORIGIN          RS  \\\n",
       "0        ISG15:9636    SO:0001583|missense_variant      1   143888043   \n",
       "1        ISG15:9636    SO:0001583|missense_variant      1   150861311   \n",
       "2        ISG15:9636            SO:0001587|nonsense      1   786201005   \n",
       "3        ISG15:9636    SO:0001583|missense_variant      1  1553169766   \n",
       "4        ISG15:9636  SO:0001819|synonymous_variant      1    61766284   \n",
       "...             ...                            ...    ...         ...   \n",
       "102316  PIK3CA:5290    SO:0001583|missense_variant      1  1060500027   \n",
       "102317  PIK3CA:5290  SO:0001819|synonymous_variant      1  1480813252   \n",
       "102318  PIK3CA:5290    SO:0001583|missense_variant      1   199563773   \n",
       "102319  PIK3CA:5290    SO:0001583|missense_variant      1   753879573   \n",
       "102320  PIK3CA:5290  SO:0001819|synonymous_variant      1   201108344   \n",
       "\n",
       "                                   CLNVI CLNSIGCONF CLNDISDBINCL CLNDNINCL  \\\n",
       "0                                    NaN        NaN          NaN       NaN   \n",
       "1                                    NaN        NaN          NaN       NaN   \n",
       "2       OMIM_Allelic_Variant:147571.0003        NaN          NaN       NaN   \n",
       "3                                    NaN        NaN          NaN       NaN   \n",
       "4                                    NaN        NaN          NaN       NaN   \n",
       "...                                  ...        ...          ...       ...   \n",
       "102316                               NaN        NaN          NaN       NaN   \n",
       "102317                               NaN        NaN          NaN       NaN   \n",
       "102318                               NaN        NaN          NaN       NaN   \n",
       "102319                               NaN        NaN          NaN       NaN   \n",
       "102320                               NaN        NaN          NaN       NaN   \n",
       "\n",
       "       CLNSIGINCL DBVARID  \n",
       "0             NaN     NaN  \n",
       "1             NaN     NaN  \n",
       "2             NaN     NaN  \n",
       "3             NaN     NaN  \n",
       "4             NaN     NaN  \n",
       "...           ...     ...  \n",
       "102316        NaN     NaN  \n",
       "102317        NaN     NaN  \n",
       "102318        NaN     NaN  \n",
       "102319        NaN     NaN  \n",
       "102320        NaN     NaN  \n",
       "\n",
       "[102321 rows x 21 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 2b: CLEAN AND MERGE DATA AS NECESSARY TO PREP FOR EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Define list of all sub-components within INFO column based on dataset documentation/meta-information\n",
    "# THIS LIST IS STRICTLY USED FOR REFERENCE\n",
    "info_scs_reflist = [\n",
    "    \"AF_ESP\",\n",
    "    \"AF_EXAC\",\n",
    "    \"AF_TGP\",\n",
    "    \"ALLELEID\",\n",
    "    \"CLNDN\",\n",
    "    \"CLNDNINCL\",\n",
    "    \"CLNDISDB\",\n",
    "    \"CLNDISDBINCL\",\n",
    "    \"CLNHGVS\",\n",
    "    \"CLNREVSTAT\",\n",
    "    \"CLNSIG\",\n",
    "    \"CLNSIGCONF\",\n",
    "    \"CLNSIGINCL\",\n",
    "    \"CLNVC\",\n",
    "    \"CLNVCSO\",\n",
    "    \"CLNVI\",\n",
    "    \"DBVARID\",\n",
    "    \"GENEINFO\",\n",
    "    \"MC\",\n",
    "    \"ORIGIN\",\n",
    "    \"RS\",\n",
    "    \"SSR\"\n",
    "]\n",
    "\n",
    "# Separate INFO column into sub-components based on consistent delimiters/characters (e.g., \";\", \"=\", etc.)\n",
    "# Transform INFO sub-components into key-value string representations in a dictionary - necessary for JSON prep\n",
    "# pd.json_normalize method takes INFO dictionary and turns it into a flat table, with sub-components as columns\n",
    "# Inspiration for parsing method (referencing here for transparency): https://www.biostars.org/p/428170/\n",
    "info_subs = '{\"' + clinical_df[\"INFO\"].str.split(';').str.join('\",\"').str.replace('=','\":\"').str.replace(\"\\\"\\\",\", \"\") + '\"}' \n",
    "clinical_info_df = pd.json_normalize(info_subs.apply(eval))\n",
    "\n",
    "# Display new dataframe containing sub-components of INFO column split out into individual columns\n",
    "clinical_info_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 2b Notes/Observations:\n",
    "- Parsing method seemed to work by splitting up INFO column into its various sub-components and storing each one as a new column\n",
    "- Only the \"SSR\" sub-component field appears to be missing, which means it must contain no entries/data\n",
    "- NaN values are apparent throughout new clinical_info_df - these values will need to be replaced with the standard missing value character (\".\") at a later step to maintain consistency with rest of dataframe\n",
    "- The new clinical_info_df should now be merged with/joined to the original clinical_info_df to bring all columns into one combined dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>AF_ESP</th>\n",
       "      <th>AF_EXAC</th>\n",
       "      <th>AF_TGP</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNHGVS</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>RS</th>\n",
       "      <th>CLNVI</th>\n",
       "      <th>CLNSIGCONF</th>\n",
       "      <th>CLNDISDBINCL</th>\n",
       "      <th>CLNDNINCL</th>\n",
       "      <th>CLNSIGINCL</th>\n",
       "      <th>DBVARID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014O42</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00546</td>\n",
       "      <td>0.00165</td>\n",
       "      <td>0.00619</td>\n",
       "      <td>446939</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014042G&gt;A</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>143888043</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1O14122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00015</td>\n",
       "      <td>0.00010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>514926</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014122C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>150861311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>181485</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014143C&gt;T</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>1</td>\n",
       "      <td>786201005</td>\n",
       "      <td>OMIM_Allelic_Variant:147571.0003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>514896</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014179C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1553169766</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.00515</td>\n",
       "      <td>0.00831</td>\n",
       "      <td>0.00339</td>\n",
       "      <td>446987</td>\n",
       "      <td>MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>NC_000001.11:g.1014217C&gt;T</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>61766284</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>3</td>\n",
       "      <td>179210507</td>\n",
       "      <td>403908</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>393412</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210507A&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1060500027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>3</td>\n",
       "      <td>179210511</td>\n",
       "      <td>526648</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>519163</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210511T&gt;C</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>1480813252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>3</td>\n",
       "      <td>179210515</td>\n",
       "      <td>526640</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>519178</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome|Hereditary_cancer-predisposing...</td>\n",
       "      <td>NC_000003.12:g.179210515A&gt;C</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>199563773</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>3</td>\n",
       "      <td>179210516</td>\n",
       "      <td>246681</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>245287</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>NC_000003.12:g.179210516A&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>753879573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>3</td>\n",
       "      <td>179210538</td>\n",
       "      <td>259958</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>251013</td>\n",
       "      <td>MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...</td>\n",
       "      <td>Cowden_syndrome|not_specified</td>\n",
       "      <td>NC_000003.12:g.179210538A&gt;T</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>SO:0001819|synonymous_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>201108344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS      ID REF ALT FILTER QUAL   AF_ESP  AF_EXAC  \\\n",
       "0           1    1014O42  475283   G   A      .    .  0.00546  0.00165   \n",
       "1           1    1O14122  542074   C   T      .    .  0.00015  0.00010   \n",
       "2           1    1014143  183381   C   T      .    .      NaN      NaN   \n",
       "3           1    1014179  542075   C   T      .    .      NaN      NaN   \n",
       "4           1    1014217  475278   C   T      .    .  0.00515  0.00831   \n",
       "...       ...        ...     ...  ..  ..    ...  ...      ...      ...   \n",
       "102316      3  179210507  403908   A   G      .    .      NaN      NaN   \n",
       "102317      3  179210511  526648   T   C      .    .      NaN      NaN   \n",
       "102318      3  179210515  526640   A   C      .    .      NaN  0.00002   \n",
       "102319      3  179210516  246681   A   G      .    .      NaN  0.00001   \n",
       "102320      3  179210538  259958   A   T      .    .      NaN  0.00001   \n",
       "\n",
       "         AF_TGP ALLELEID                                           CLNDISDB  \\\n",
       "0       0.00619   446939   MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "1           NaN   514926   MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "2           NaN   181485   MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "3           NaN   514896   MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "4       0.00339   446987   MedGen:C4015293,OMIM:616126,Orphanet:ORPHA319563   \n",
       "...         ...      ...                                                ...   \n",
       "102316      NaN   393412  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102317      NaN   519163  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102318      NaN   519178  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102319      NaN   245287  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "102320      NaN   251013  MedGen:C0018553,Orphanet:ORPHA201,SNOMED_CT:58...   \n",
       "\n",
       "                                                    CLNDN  \\\n",
       "0       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "1       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "2       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "3       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "4       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "...                                                   ...   \n",
       "102316                                    Cowden_syndrome   \n",
       "102317                                    Cowden_syndrome   \n",
       "102318  Cowden_syndrome|Hereditary_cancer-predisposing...   \n",
       "102319                                    Cowden_syndrome   \n",
       "102320                      Cowden_syndrome|not_specified   \n",
       "\n",
       "                            CLNHGVS  \\\n",
       "0         NC_000001.11:g.1014042G>A   \n",
       "1         NC_000001.11:g.1014122C>T   \n",
       "2         NC_000001.11:g.1014143C>T   \n",
       "3         NC_000001.11:g.1014179C>T   \n",
       "4         NC_000001.11:g.1014217C>T   \n",
       "...                             ...   \n",
       "102316  NC_000003.12:g.179210507A>G   \n",
       "102317  NC_000003.12:g.179210511T>C   \n",
       "102318  NC_000003.12:g.179210515A>C   \n",
       "102319  NC_000003.12:g.179210516A>G   \n",
       "102320  NC_000003.12:g.179210538A>T   \n",
       "\n",
       "                                               CLNREVSTAT  \\\n",
       "0                     criteria_provided,_single_submitter   \n",
       "1                     criteria_provided,_single_submitter   \n",
       "2                          no_assertion_criteria_provided   \n",
       "3                     criteria_provided,_single_submitter   \n",
       "4                     criteria_provided,_single_submitter   \n",
       "...                                                   ...   \n",
       "102316                criteria_provided,_single_submitter   \n",
       "102317                criteria_provided,_single_submitter   \n",
       "102318  criteria_provided,_multiple_submitters,_no_con...   \n",
       "102319                criteria_provided,_single_submitter   \n",
       "102320  criteria_provided,_multiple_submitters,_no_con...   \n",
       "\n",
       "                        CLNSIG                      CLNVC     CLNVCSO  \\\n",
       "0                       Benign  single_nucleotide_variant  SO:0001483   \n",
       "1       Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "2                   Pathogenic  single_nucleotide_variant  SO:0001483   \n",
       "3       Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "4                       Benign  single_nucleotide_variant  SO:0001483   \n",
       "...                        ...                        ...         ...   \n",
       "102316  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102317           Likely_benign  single_nucleotide_variant  SO:0001483   \n",
       "102318  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102319  Uncertain_significance  single_nucleotide_variant  SO:0001483   \n",
       "102320           Likely_benign  single_nucleotide_variant  SO:0001483   \n",
       "\n",
       "           GENEINFO                             MC ORIGIN          RS  \\\n",
       "0        ISG15:9636    SO:0001583|missense_variant      1   143888043   \n",
       "1        ISG15:9636    SO:0001583|missense_variant      1   150861311   \n",
       "2        ISG15:9636            SO:0001587|nonsense      1   786201005   \n",
       "3        ISG15:9636    SO:0001583|missense_variant      1  1553169766   \n",
       "4        ISG15:9636  SO:0001819|synonymous_variant      1    61766284   \n",
       "...             ...                            ...    ...         ...   \n",
       "102316  PIK3CA:5290    SO:0001583|missense_variant      1  1060500027   \n",
       "102317  PIK3CA:5290  SO:0001819|synonymous_variant      1  1480813252   \n",
       "102318  PIK3CA:5290    SO:0001583|missense_variant      1   199563773   \n",
       "102319  PIK3CA:5290    SO:0001583|missense_variant      1   753879573   \n",
       "102320  PIK3CA:5290  SO:0001819|synonymous_variant      1   201108344   \n",
       "\n",
       "                                   CLNVI CLNSIGCONF CLNDISDBINCL CLNDNINCL  \\\n",
       "0                                    NaN        NaN          NaN       NaN   \n",
       "1                                    NaN        NaN          NaN       NaN   \n",
       "2       OMIM_Allelic_Variant:147571.0003        NaN          NaN       NaN   \n",
       "3                                    NaN        NaN          NaN       NaN   \n",
       "4                                    NaN        NaN          NaN       NaN   \n",
       "...                                  ...        ...          ...       ...   \n",
       "102316                               NaN        NaN          NaN       NaN   \n",
       "102317                               NaN        NaN          NaN       NaN   \n",
       "102318                               NaN        NaN          NaN       NaN   \n",
       "102319                               NaN        NaN          NaN       NaN   \n",
       "102320                               NaN        NaN          NaN       NaN   \n",
       "\n",
       "       CLNSIGINCL DBVARID  \n",
       "0             NaN     NaN  \n",
       "1             NaN     NaN  \n",
       "2             NaN     NaN  \n",
       "3             NaN     NaN  \n",
       "4             NaN     NaN  \n",
       "...           ...     ...  \n",
       "102316        NaN     NaN  \n",
       "102317        NaN     NaN  \n",
       "102318        NaN     NaN  \n",
       "102319        NaN     NaN  \n",
       "102320        NaN     NaN  \n",
       "\n",
       "[102321 rows x 28 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 2c: CLEAN AND MERGE DATA AS NECESSARY TO PREP FOR EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Merge original clinical_df and clinical_info_df together (inner join - 1:1 index match)\n",
    "clinical_expanded_df = pd.merge(\n",
    "    clinical_df,\n",
    "    clinical_info_df,\n",
    "    left_index = True,\n",
    "    right_index = True\n",
    ")\n",
    "\n",
    "# Drop original INFO column now that all INFO sub-components are included in clinical_expanded_df\n",
    "clinical_expanded_df.drop([\"INFO\"], axis = 1, inplace = True)\n",
    "\n",
    "# Display new merged dataframe\n",
    "clinical_expanded_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 2c Notes/Observations:\n",
    "- Merging clinical_df and clinical_info_df worked exactly as expected and brought in all sub-components/sub-fields from the INFO column\n",
    "- Original INFO column in new clinical_expanded_df was redundant after the merge, so it was dropped\n",
    "- Now that we have a relatively clean and organized dataframe, this would be a good time to update all missing values (\".\" and NaN) with \"Not_Given\" - per the instructions above - and update all other improper characters in additional columns (e.g., letter characters in the POS column) as necessary. This will help get the dataset ready for additional column-specific deep-dives and analysis (e.g., value_counts/distributions of values, etc.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>GENEINFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>ISG15:9636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>ISG15:9636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ISG15:9636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>ISG15:9636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>ISG15:9636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>3</td>\n",
       "      <td>179210507</td>\n",
       "      <td>403908</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>3</td>\n",
       "      <td>179210511</td>\n",
       "      <td>526648</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>3</td>\n",
       "      <td>179210515</td>\n",
       "      <td>526640</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Cowden_syndrome|Hereditary_cancer-predisposing...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>3</td>\n",
       "      <td>179210516</td>\n",
       "      <td>246681</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>3</td>\n",
       "      <td>179210538</td>\n",
       "      <td>259958</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Cowden_syndrome|not_specified</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS      ID REF ALT  \\\n",
       "0           1    1014042  475283   G   A   \n",
       "1           1    1014122  542074   C   T   \n",
       "2           1    1014143  183381   C   T   \n",
       "3           1    1014179  542075   C   T   \n",
       "4           1    1014217  475278   C   T   \n",
       "...       ...        ...     ...  ..  ..   \n",
       "102316      3  179210507  403908   A   G   \n",
       "102317      3  179210511  526648   T   C   \n",
       "102318      3  179210515  526640   A   C   \n",
       "102319      3  179210516  246681   A   G   \n",
       "102320      3  179210538  259958   A   T   \n",
       "\n",
       "                                                    CLNDN  \\\n",
       "0       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "1       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "2       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "3       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "4       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "...                                                   ...   \n",
       "102316                                    Cowden_syndrome   \n",
       "102317                                    Cowden_syndrome   \n",
       "102318  Cowden_syndrome|Hereditary_cancer-predisposing...   \n",
       "102319                                    Cowden_syndrome   \n",
       "102320                      Cowden_syndrome|not_specified   \n",
       "\n",
       "                        CLNSIG     GENEINFO  \n",
       "0                       Benign   ISG15:9636  \n",
       "1       Uncertain_significance   ISG15:9636  \n",
       "2                   Pathogenic   ISG15:9636  \n",
       "3       Uncertain_significance   ISG15:9636  \n",
       "4                       Benign   ISG15:9636  \n",
       "...                        ...          ...  \n",
       "102316  Uncertain_significance  PIK3CA:5290  \n",
       "102317           Likely_benign  PIK3CA:5290  \n",
       "102318  Uncertain_significance  PIK3CA:5290  \n",
       "102319  Uncertain_significance  PIK3CA:5290  \n",
       "102320           Likely_benign  PIK3CA:5290  \n",
       "\n",
       "[102321 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 2d: CLEAN AND MERGE DATA AS NECESSARY TO PREP FOR EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Replace all NaN values and standard missing values (\".\") in dataframe with \"Not_Given\"\n",
    "clinical_expanded_df.replace(\n",
    "    {\n",
    "        np.nan: \"Not_Given\",\n",
    "        \".\": \"Not_Given\"\n",
    "    },\n",
    "    inplace = True\n",
    ")\n",
    "\n",
    "# Replace \"O\" characters in \"POS\" column with \"0\", since column is supposed to only include integer values\n",
    "clinical_expanded_df[\"POS\"] = clinical_expanded_df[\"POS\"].str.replace(\"O\", \"0\")\n",
    "\n",
    "# Subset clinical_expanded_df to only columns of interest, filtering out any unnecessary/extraneous columns\n",
    "clinical_filtered_df = clinical_expanded_df[\n",
    "    [\n",
    "        \"CHROM\",\n",
    "        \"POS\",\n",
    "        \"ID\",\n",
    "        \"REF\",\n",
    "        \"ALT\",\n",
    "        \"CLNDN\",\n",
    "        \"CLNSIG\",\n",
    "        \"GENEINFO\"\n",
    "    ]\n",
    "]\n",
    "\n",
    "# Display filtered dataframe (post-cleaning)\n",
    "clinical_filtered_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 2d Notes/Observations:\n",
    "- Updates to dataframe worked successfully - all missing/NaN entries now show \"Not_Given\" as expected and POS column now shows only numbers (instead of strings with \"O\" characters in them)\n",
    "- clinical_expanded_df now appears to be clean and ready for further investigation/analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataframe info:\n",
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 102321 entries, 0 to 102320\n",
      "Data columns (total 8 columns):\n",
      " #   Column    Non-Null Count   Dtype \n",
      "---  ------    --------------   ----- \n",
      " 0   CHROM     102321 non-null  int64 \n",
      " 1   POS       102321 non-null  object\n",
      " 2   ID        102321 non-null  int64 \n",
      " 3   REF       102321 non-null  object\n",
      " 4   ALT       102321 non-null  object\n",
      " 5   CLNDN     102321 non-null  object\n",
      " 6   CLNSIG    102321 non-null  object\n",
      " 7   GENEINFO  102321 non-null  object\n",
      "dtypes: int64(2), object(6)\n",
      "memory usage: 6.2+ MB\n",
      "None\n",
      "\n",
      "Dataframe shape: (102321, 8)\n",
      "\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Distribution of Variants by Clinical Significance:\n",
      "Uncertain_significance                                                                      47980\n",
      "Likely_benign                                                                               17885\n",
      "Pathogenic                                                                                  12313\n",
      "Likely_pathogenic                                                                            6269\n",
      "Benign                                                                                       6138\n",
      "Conflicting_interpretations_of_pathogenicity                                                 5404\n",
      "Benign/Likely_benign                                                                         3338\n",
      "Not_Given                                                                                    1797\n",
      "Pathogenic/Likely_pathogenic                                                                  854\n",
      "risk_factor                                                                                    98\n",
      "association                                                                                    70\n",
      "drug_response                                                                                  41\n",
      "Affects                                                                                        31\n",
      "Conflicting_interpretations_of_pathogenicity,_risk_factor                                      14\n",
      "other                                                                                          12\n",
      "Pathogenic,_risk_factor                                                                        11\n",
      "Conflicting_interpretations_of_pathogenicity,_other                                             8\n",
      "Pathogenic/Likely_pathogenic,_other                                                             8\n",
      "Benign/Likely_benign,_risk_factor                                                               7\n",
      "protective                                                                                      6\n",
      "Pathogenic/Likely_pathogenic,_risk_factor                                                       6\n",
      "Likely_pathogenic,_risk_factor                                                                  5\n",
      "protective,_risk_factor                                                                         4\n",
      "Pathogenic,_other                                                                               2\n",
      "Pathogenic,_Affects                                                                             2\n",
      "Uncertain_significance,_risk_factor                                                             1\n",
      "Benign/Likely_benign,_protective                                                                1\n",
      "Affects,_risk_factor                                                                            1\n",
      "Conflicting_interpretations_of_pathogenicity,_protective                                        1\n",
      "Likely_pathogenic,_other                                                                        1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects,_other                                    1\n",
      "Benign,_drug_response                                                                           1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects,_association,_drug_response,_other        1\n",
      "Pathogenic,_protective                                                                          1\n",
      "Benign,_other                                                                                   1\n",
      "Likely_pathogenic,_association                                                                  1\n",
      "Conflicting_interpretations_of_pathogenicity,_Affects                                           1\n",
      "Pathogenic,_association,_protective                                                             1\n",
      "Likely_benign,_other                                                                            1\n",
      "Benign/Likely_benign,_other                                                                     1\n",
      "Conflicting_interpretations_of_pathogenicity,_other,_risk_factor                                1\n",
      "Conflicting_interpretations_of_pathogenicity,_association                                       1\n",
      "Benign/Likely_benign,_association                                                               1\n",
      "Name: CLNSIG, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 3a: PERFORM INITIAL/BASIC EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Get basic info, Dtypes, and shape for updated/cleaned dataframe to ensure no rows have dropped along the way\n",
    "print(\"Dataframe info:\")\n",
    "print(clinical_filtered_df.info(), end = \"\\n\\n\")\n",
    "print(\"Dataframe shape:\", clinical_filtered_df.shape, end = \"\\n\\n\")\n",
    "print(\"-\" * 100)\n",
    "\n",
    "# Display distribution (counts) of variants by clinical significance to see unique/highest-density CLNSIG values\n",
    "print(\"Distribution of Variants by Clinical Significance:\")\n",
    "print(clinical_filtered_df[\"CLNSIG\"].value_counts())\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 3a Notes/Observations:\n",
    "- No rows were dropped in the cleaning and filtering steps above (good)\n",
    "- Distribution of variant counts (rows of data) by clinical significance category provides a helpful view:\n",
    "    - Almost half (~50%) of all variants (rows of data) have the \"Uncertain_significance\" label for CLNSIG\n",
    "    - The remaining variants (rows of data) are distributed across a variety of CLNSIG categories, with many containing labels for \"pathogenic\" or \"benign\" (though not all)\n",
    "    - These labels will serve as the starting point for mapping variants into a simple binary classification: \"dangerous\" versus \"all other\" (which would include both non-dangerous/benign variants and variants with inconclusive or conflicting significance)\n",
    "- A good next step will be to collapse this data down to a simpler, binary (0 vs. 1) variable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/8p/70n8s0n94xq5btk8hg6g4n3c0000gn/T/ipykernel_27777/2690195186.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  clinical_filtered_df[\"IMPLIEDCLASS\"] = clinical_filtered_df[\"CLNSIG\"].map(\n",
      "/opt/anaconda3/lib/python3.8/site-packages/pandas/core/generic.py:6619: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return self._update_inplace(result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>IMPLIEDCLASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>475283</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>542074</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>542075</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>475278</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Benign</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102316</th>\n",
       "      <td>3</td>\n",
       "      <td>179210507</td>\n",
       "      <td>403908</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102317</th>\n",
       "      <td>3</td>\n",
       "      <td>179210511</td>\n",
       "      <td>526648</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102318</th>\n",
       "      <td>3</td>\n",
       "      <td>179210515</td>\n",
       "      <td>526640</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Cowden_syndrome|Hereditary_cancer-predisposing...</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102319</th>\n",
       "      <td>3</td>\n",
       "      <td>179210516</td>\n",
       "      <td>246681</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Cowden_syndrome</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102320</th>\n",
       "      <td>3</td>\n",
       "      <td>179210538</td>\n",
       "      <td>259958</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Cowden_syndrome|not_specified</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>PIK3CA:5290</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102321 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM        POS      ID REF ALT  \\\n",
       "0           1    1014042  475283   G   A   \n",
       "1           1    1014122  542074   C   T   \n",
       "2           1    1014143  183381   C   T   \n",
       "3           1    1014179  542075   C   T   \n",
       "4           1    1014217  475278   C   T   \n",
       "...       ...        ...     ...  ..  ..   \n",
       "102316      3  179210507  403908   A   G   \n",
       "102317      3  179210511  526648   T   C   \n",
       "102318      3  179210515  526640   A   C   \n",
       "102319      3  179210516  246681   A   G   \n",
       "102320      3  179210538  259958   A   T   \n",
       "\n",
       "                                                    CLNDN  \\\n",
       "0       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "1       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "2       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "3       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "4       Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "...                                                   ...   \n",
       "102316                                    Cowden_syndrome   \n",
       "102317                                    Cowden_syndrome   \n",
       "102318  Cowden_syndrome|Hereditary_cancer-predisposing...   \n",
       "102319                                    Cowden_syndrome   \n",
       "102320                      Cowden_syndrome|not_specified   \n",
       "\n",
       "                        CLNSIG     GENEINFO  IMPLIEDCLASS  \n",
       "0                       Benign   ISG15:9636          0.00  \n",
       "1       Uncertain_significance   ISG15:9636          0.00  \n",
       "2                   Pathogenic   ISG15:9636          1.00  \n",
       "3       Uncertain_significance   ISG15:9636          0.00  \n",
       "4                       Benign   ISG15:9636          0.00  \n",
       "...                        ...          ...           ...  \n",
       "102316  Uncertain_significance  PIK3CA:5290          0.00  \n",
       "102317           Likely_benign  PIK3CA:5290          0.00  \n",
       "102318  Uncertain_significance  PIK3CA:5290          0.00  \n",
       "102319  Uncertain_significance  PIK3CA:5290          0.00  \n",
       "102320           Likely_benign  PIK3CA:5290          0.00  \n",
       "\n",
       "[102321 rows x 9 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 3b: PERFORM INITIAL/BASIC EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Create new IMPLIEDCLASS column in dataframe by mapping certain CLNSIG values to 1s\n",
    "# This new column assigns a value of 1 to presumed \"dangerous\" variants (defined below)\n",
    "clinical_filtered_df[\"IMPLIEDCLASS\"] = clinical_filtered_df[\"CLNSIG\"].map(\n",
    "    {\n",
    "        \"Pathogenic\": 1,\n",
    "        \"Likely_pathogenic\": 1,\n",
    "        \"Pathogenic/Likely_pathogenic\": 1,\n",
    "        \"risk_factor\": 1,\n",
    "        \"Pathogenic,_risk_factor\": 1,\n",
    "        \"Pathogenic/Likely_pathogenic,_other\": 1,\n",
    "        \"Pathogenic/Likely_pathogenic,_risk_factor\": 1,\n",
    "        \"Likely_pathogenic,_risk_factor\": 1,\n",
    "        \"Pathogenic,_other\": 1,\n",
    "        \"Pathogenic,_Affects\": 1,\n",
    "        \"Likely_pathogenic,_other\": 1,\n",
    "        \"Pathogenic,_protective\": 1,\n",
    "        \"Likely_pathogenic,_association\": 1,\n",
    "        \"Pathogenic,_association,_protective\": 1\n",
    "    }\n",
    ")\n",
    "\n",
    "# Replace all NaN values in new IMPLIEDCLASS column with 0s\n",
    "clinical_filtered_df[\"IMPLIEDCLASS\"].replace(\n",
    "    {np.nan: 0},\n",
    "    inplace = True\n",
    ")\n",
    "\n",
    "# Display new version of dataframe with additional column\n",
    "clinical_filtered_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Step 3b Notes/Observations:\n",
    "- New IMPLIEDCLASS column now categorizes variants as potentially \"dangerous\" (1) or \"non-dangerous/benign/inconclusive\" (0) for easy grouping and summarizing\n",
    "- A good next step will be to determine what percentage of variants (rows) in the dataframe are potentially dangerous/harmful based upon this categorization (i.e., variants labeled with 1 only), and then to limit the dataframe to the first 100 potentially dangerous/harmful variants per the instructions "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of Potentially Dangerous/Harmful Variants (IMPLIEDCLASS = 1):\n",
      "19572.0\n",
      "\n",
      "Percentage of Potentially Dangerous/Harmful Variants out of Total Variants:\n",
      "19.0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>IMPLIEDCLASS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>183381</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1014316</td>\n",
       "      <td>161455</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1014359</td>\n",
       "      <td>161454</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Immunodeficiency_38_with_basal_ganglia_calcifi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ISG15:9636</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>243036</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>1022313</td>\n",
       "      <td>243037</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>46</td>\n",
       "      <td>1</td>\n",
       "      <td>1041354</td>\n",
       "      <td>574478</td>\n",
       "      <td>CGCCCGCCAGGAGAATGTCTTCAAGAAGTTCGACG</td>\n",
       "      <td>C</td>\n",
       "      <td>Myasthenic_syndrome,_congenital,_8</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>126556</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>63</td>\n",
       "      <td>1</td>\n",
       "      <td>1042136</td>\n",
       "      <td>243038</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>237</td>\n",
       "      <td>1</td>\n",
       "      <td>1049672</td>\n",
       "      <td>489335</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>270</td>\n",
       "      <td>1</td>\n",
       "      <td>1050473</td>\n",
       "      <td>243039</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>280</td>\n",
       "      <td>1</td>\n",
       "      <td>1050575</td>\n",
       "      <td>18241</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>282</td>\n",
       "      <td>1</td>\n",
       "      <td>1050763</td>\n",
       "      <td>126555</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Congenital_myasthenic_syndrome|Myasthenic_synd...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>292</td>\n",
       "      <td>1</td>\n",
       "      <td>1051309</td>\n",
       "      <td>503670</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>314</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>243040</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Congenital_myasthenic_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>AGRN:375790</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>365</td>\n",
       "      <td>1</td>\n",
       "      <td>1213738</td>\n",
       "      <td>96692</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Immunodeficiency_16</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TNFRSF4:7293</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>371</td>\n",
       "      <td>1</td>\n",
       "      <td>1232279</td>\n",
       "      <td>60484</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>375</td>\n",
       "      <td>1</td>\n",
       "      <td>1232294</td>\n",
       "      <td>60493</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>387</td>\n",
       "      <td>1</td>\n",
       "      <td>1232471</td>\n",
       "      <td>60488</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>388</td>\n",
       "      <td>1</td>\n",
       "      <td>1232478</td>\n",
       "      <td>60489</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>390</td>\n",
       "      <td>1</td>\n",
       "      <td>1232517</td>\n",
       "      <td>452846</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>391</td>\n",
       "      <td>1</td>\n",
       "      <td>1232594</td>\n",
       "      <td>60497</td>\n",
       "      <td>CGGCGCGCCCTGGAGCGGGAGCA</td>\n",
       "      <td>C</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>393</td>\n",
       "      <td>1</td>\n",
       "      <td>1232630</td>\n",
       "      <td>60490</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>398</td>\n",
       "      <td>1</td>\n",
       "      <td>1232684</td>\n",
       "      <td>60494</td>\n",
       "      <td>GTGCTGGCCA</td>\n",
       "      <td>G</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>401</td>\n",
       "      <td>1</td>\n",
       "      <td>1232744</td>\n",
       "      <td>60486</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>406</td>\n",
       "      <td>1</td>\n",
       "      <td>1232834</td>\n",
       "      <td>522415</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>407</td>\n",
       "      <td>1</td>\n",
       "      <td>1232859</td>\n",
       "      <td>427130</td>\n",
       "      <td>C</td>\n",
       "      <td>CG</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>408</td>\n",
       "      <td>1</td>\n",
       "      <td>1232859</td>\n",
       "      <td>60492</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>414</td>\n",
       "      <td>1</td>\n",
       "      <td>1232897</td>\n",
       "      <td>60495</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>416</td>\n",
       "      <td>1</td>\n",
       "      <td>1232927</td>\n",
       "      <td>60496</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>418</td>\n",
       "      <td>1</td>\n",
       "      <td>1232972</td>\n",
       "      <td>60485</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>425</td>\n",
       "      <td>1</td>\n",
       "      <td>1233177</td>\n",
       "      <td>60487</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Spondyloepimetaphyseal_dysplasia_with_joint_la...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>427</td>\n",
       "      <td>1</td>\n",
       "      <td>1233203</td>\n",
       "      <td>60491</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Ehlers-Danlos_syndrome,_progeroid_type,_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>B3GALT6:126792</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>430</td>\n",
       "      <td>1</td>\n",
       "      <td>1336498</td>\n",
       "      <td>504195</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>431</td>\n",
       "      <td>1</td>\n",
       "      <td>1337984</td>\n",
       "      <td>524041</td>\n",
       "      <td>CTGCTGACTCCCGG</td>\n",
       "      <td>C</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>432</td>\n",
       "      <td>1</td>\n",
       "      <td>1337992</td>\n",
       "      <td>488046</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>433</td>\n",
       "      <td>1</td>\n",
       "      <td>1337992</td>\n",
       "      <td>488048</td>\n",
       "      <td>TCCCGGTGCTGCCGCTG</td>\n",
       "      <td>T</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>434</td>\n",
       "      <td>1</td>\n",
       "      <td>1337999</td>\n",
       "      <td>488045</td>\n",
       "      <td>G</td>\n",
       "      <td>GCTGCC</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>435</td>\n",
       "      <td>1</td>\n",
       "      <td>1338000</td>\n",
       "      <td>208047</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>436</td>\n",
       "      <td>1</td>\n",
       "      <td>1338033</td>\n",
       "      <td>208050</td>\n",
       "      <td>TAGGCAGG</td>\n",
       "      <td>C</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>437</td>\n",
       "      <td>1</td>\n",
       "      <td>1338045</td>\n",
       "      <td>208043</td>\n",
       "      <td>AA</td>\n",
       "      <td>G</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>438</td>\n",
       "      <td>1</td>\n",
       "      <td>1338053</td>\n",
       "      <td>208049</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>439</td>\n",
       "      <td>1</td>\n",
       "      <td>1338086</td>\n",
       "      <td>208048</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>440</td>\n",
       "      <td>1</td>\n",
       "      <td>1338092</td>\n",
       "      <td>219223</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>441</td>\n",
       "      <td>1</td>\n",
       "      <td>1338096</td>\n",
       "      <td>208045</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>442</td>\n",
       "      <td>1</td>\n",
       "      <td>1338098</td>\n",
       "      <td>208044</td>\n",
       "      <td>GGGGGCAGCCGGGT</td>\n",
       "      <td>G</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>443</td>\n",
       "      <td>1</td>\n",
       "      <td>1338107</td>\n",
       "      <td>208046</td>\n",
       "      <td>CG</td>\n",
       "      <td>C</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>444</td>\n",
       "      <td>1</td>\n",
       "      <td>1338107</td>\n",
       "      <td>488047</td>\n",
       "      <td>CGGGTGGGGCAGCG</td>\n",
       "      <td>C</td>\n",
       "      <td>Robinow_syndrome,_autosomal_dominant_2</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>445</td>\n",
       "      <td>1</td>\n",
       "      <td>1338111</td>\n",
       "      <td>520999</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>448</td>\n",
       "      <td>1</td>\n",
       "      <td>1338529</td>\n",
       "      <td>373812</td>\n",
       "      <td>G</td>\n",
       "      <td>GGCATTGGC</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>DVL1:1855</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>452</td>\n",
       "      <td>1</td>\n",
       "      <td>1353913</td>\n",
       "      <td>402162</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Abnormality_of_brain_morphology</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>MXRA8:54587</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>457</td>\n",
       "      <td>1</td>\n",
       "      <td>1517250</td>\n",
       "      <td>592109</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>Harel-Yoon_syndrome</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>464</td>\n",
       "      <td>1</td>\n",
       "      <td>1529299</td>\n",
       "      <td>225696</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Harel-Yoon_syndrome|not_provided</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>465</td>\n",
       "      <td>1</td>\n",
       "      <td>1534004</td>\n",
       "      <td>522903</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Harel-Yoon_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>ATAD3A:55210</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>466</td>\n",
       "      <td>1</td>\n",
       "      <td>1535370</td>\n",
       "      <td>161196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>467</td>\n",
       "      <td>1</td>\n",
       "      <td>1535372</td>\n",
       "      <td>161192</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spinocerebellar_ataxia_21|not_provided</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>468</td>\n",
       "      <td>1</td>\n",
       "      <td>1535392</td>\n",
       "      <td>161193</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>472</td>\n",
       "      <td>1</td>\n",
       "      <td>1535616</td>\n",
       "      <td>161194</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>474</td>\n",
       "      <td>1</td>\n",
       "      <td>1535619</td>\n",
       "      <td>523023</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>475</td>\n",
       "      <td>1</td>\n",
       "      <td>1535723</td>\n",
       "      <td>161195</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Spinocerebellar_ataxia_21</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>476</td>\n",
       "      <td>1</td>\n",
       "      <td>1535766</td>\n",
       "      <td>372833</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>TMEM240:339453</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>480</td>\n",
       "      <td>1</td>\n",
       "      <td>1787340</td>\n",
       "      <td>521025</td>\n",
       "      <td>GATC</td>\n",
       "      <td>G</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>481</td>\n",
       "      <td>1</td>\n",
       "      <td>1787378</td>\n",
       "      <td>224718</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>484</td>\n",
       "      <td>1</td>\n",
       "      <td>1804461</td>\n",
       "      <td>452797</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>485</td>\n",
       "      <td>1</td>\n",
       "      <td>1804496</td>\n",
       "      <td>521213</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Inborn_genetic_diseases</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>487</td>\n",
       "      <td>1</td>\n",
       "      <td>1804548</td>\n",
       "      <td>224717</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>488</td>\n",
       "      <td>1</td>\n",
       "      <td>1804565</td>\n",
       "      <td>224716</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>489</td>\n",
       "      <td>1</td>\n",
       "      <td>1806476</td>\n",
       "      <td>422536</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>490</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>391609</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>491</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>208722</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Cleft_...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>492</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>224715</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Seizur...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>493</td>\n",
       "      <td>1</td>\n",
       "      <td>1806509</td>\n",
       "      <td>224714</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>494</td>\n",
       "      <td>1</td>\n",
       "      <td>1806512</td>\n",
       "      <td>431082</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Mental_retardation,_autosomal_dominant_42</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>495</td>\n",
       "      <td>1</td>\n",
       "      <td>1806513</td>\n",
       "      <td>224713</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Seizur...</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>496</td>\n",
       "      <td>1</td>\n",
       "      <td>1806514</td>\n",
       "      <td>224712</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Strabi...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>497</td>\n",
       "      <td>1</td>\n",
       "      <td>1806515</td>\n",
       "      <td>224711</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Neurodevelopmental_Disability|hypotonia|Intell...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>498</td>\n",
       "      <td>1</td>\n",
       "      <td>1815801</td>\n",
       "      <td>617622</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Mental_retardation,_autosomal_dominant_42</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>GNB1:2782</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>519</td>\n",
       "      <td>1</td>\n",
       "      <td>2027636</td>\n",
       "      <td>16212</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Generalized_epilepsy_with_febrile_seizures_plu...</td>\n",
       "      <td>risk_factor</td>\n",
       "      <td>GABRD:2563</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>568</td>\n",
       "      <td>1</td>\n",
       "      <td>2228828</td>\n",
       "      <td>224869</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>569</td>\n",
       "      <td>1</td>\n",
       "      <td>2228834</td>\n",
       "      <td>570752</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>571</td>\n",
       "      <td>1</td>\n",
       "      <td>2228860</td>\n",
       "      <td>37259</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>573</td>\n",
       "      <td>1</td>\n",
       "      <td>2228866</td>\n",
       "      <td>37261</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Inborn_genetic_diseases|Shprintzen-Goldberg_sy...</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>574</td>\n",
       "      <td>1</td>\n",
       "      <td>2228866</td>\n",
       "      <td>37262</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Inborn_genetic_diseases|Shprintzen-Goldberg_sy...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>575</td>\n",
       "      <td>1</td>\n",
       "      <td>2228867</td>\n",
       "      <td>37260</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>576</td>\n",
       "      <td>1</td>\n",
       "      <td>2228867</td>\n",
       "      <td>39783</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>578</td>\n",
       "      <td>1</td>\n",
       "      <td>2228870</td>\n",
       "      <td>39785</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>599</td>\n",
       "      <td>1</td>\n",
       "      <td>2229045</td>\n",
       "      <td>37263</td>\n",
       "      <td>GTCCGACCGC</td>\n",
       "      <td>G</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>600</td>\n",
       "      <td>1</td>\n",
       "      <td>2229045</td>\n",
       "      <td>39784</td>\n",
       "      <td>GTCCGACCGCTCC</td>\n",
       "      <td>G</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>605</td>\n",
       "      <td>1</td>\n",
       "      <td>2229113</td>\n",
       "      <td>37258</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>606</td>\n",
       "      <td>1</td>\n",
       "      <td>2229115</td>\n",
       "      <td>224868</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome|not_provided</td>\n",
       "      <td>Pathogenic/Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>607</td>\n",
       "      <td>1</td>\n",
       "      <td>2229118</td>\n",
       "      <td>222819</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>622</td>\n",
       "      <td>1</td>\n",
       "      <td>2229265</td>\n",
       "      <td>213703</td>\n",
       "      <td>CT</td>\n",
       "      <td>GG</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>626</td>\n",
       "      <td>1</td>\n",
       "      <td>2229305</td>\n",
       "      <td>495053</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Disproportionate_tall_stature</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>634</td>\n",
       "      <td>1</td>\n",
       "      <td>2229401</td>\n",
       "      <td>463409</td>\n",
       "      <td>AGCTCAGCGAGCGCAGCGTCCGCGTGTACCACGAGTGCTTCGGCAA...</td>\n",
       "      <td>A</td>\n",
       "      <td>Shprintzen-Goldberg_syndrome</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>SKI:6497</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>846</td>\n",
       "      <td>1</td>\n",
       "      <td>2406483</td>\n",
       "      <td>554342</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>847</td>\n",
       "      <td>1</td>\n",
       "      <td>2406483</td>\n",
       "      <td>557020</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>850</td>\n",
       "      <td>1</td>\n",
       "      <td>2406526</td>\n",
       "      <td>6771</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6B</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>852</td>\n",
       "      <td>1</td>\n",
       "      <td>2406528</td>\n",
       "      <td>197886</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>857</td>\n",
       "      <td>1</td>\n",
       "      <td>2406561</td>\n",
       "      <td>282334</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Not_Given</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>862</td>\n",
       "      <td>1</td>\n",
       "      <td>2406580</td>\n",
       "      <td>555297</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>Peroxisome_biogenesis_disorder_6A|Peroxisome_b...</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>863</td>\n",
       "      <td>1</td>\n",
       "      <td>2406580</td>\n",
       "      <td>296273</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>Zellweger_syndrome|Peroxisome_biogenesis_disor...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PEX10:5192</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index  CHROM      POS      ID  \\\n",
       "0       2      1  1014143  183381   \n",
       "1       8      1  1014316  161455   \n",
       "2       9      1  1014359  161454   \n",
       "3      24      1  1022225  243036   \n",
       "4      26      1  1022313  243037   \n",
       "5      46      1  1041354  574478   \n",
       "6      49      1  1041582  126556   \n",
       "7      63      1  1042136  243038   \n",
       "8     237      1  1049672  489335   \n",
       "9     270      1  1050473  243039   \n",
       "10    280      1  1050575   18241   \n",
       "11    282      1  1050763  126555   \n",
       "12    292      1  1051309  503670   \n",
       "13    314      1  1051775  243040   \n",
       "14    365      1  1213738   96692   \n",
       "15    371      1  1232279   60484   \n",
       "16    375      1  1232294   60493   \n",
       "17    387      1  1232471   60488   \n",
       "18    388      1  1232478   60489   \n",
       "19    390      1  1232517  452846   \n",
       "20    391      1  1232594   60497   \n",
       "21    393      1  1232630   60490   \n",
       "22    398      1  1232684   60494   \n",
       "23    401      1  1232744   60486   \n",
       "24    406      1  1232834  522415   \n",
       "25    407      1  1232859  427130   \n",
       "26    408      1  1232859   60492   \n",
       "27    414      1  1232897   60495   \n",
       "28    416      1  1232927   60496   \n",
       "29    418      1  1232972   60485   \n",
       "30    425      1  1233177   60487   \n",
       "31    427      1  1233203   60491   \n",
       "32    430      1  1336498  504195   \n",
       "33    431      1  1337984  524041   \n",
       "34    432      1  1337992  488046   \n",
       "35    433      1  1337992  488048   \n",
       "36    434      1  1337999  488045   \n",
       "37    435      1  1338000  208047   \n",
       "38    436      1  1338033  208050   \n",
       "39    437      1  1338045  208043   \n",
       "40    438      1  1338053  208049   \n",
       "41    439      1  1338086  208048   \n",
       "42    440      1  1338092  219223   \n",
       "43    441      1  1338096  208045   \n",
       "44    442      1  1338098  208044   \n",
       "45    443      1  1338107  208046   \n",
       "46    444      1  1338107  488047   \n",
       "47    445      1  1338111  520999   \n",
       "48    448      1  1338529  373812   \n",
       "49    452      1  1353913  402162   \n",
       "50    457      1  1517250  592109   \n",
       "51    464      1  1529299  225696   \n",
       "52    465      1  1534004  522903   \n",
       "53    466      1  1535370  161196   \n",
       "54    467      1  1535372  161192   \n",
       "55    468      1  1535392  161193   \n",
       "56    472      1  1535616  161194   \n",
       "57    474      1  1535619  523023   \n",
       "58    475      1  1535723  161195   \n",
       "59    476      1  1535766  372833   \n",
       "60    480      1  1787340  521025   \n",
       "61    481      1  1787378  224718   \n",
       "62    484      1  1804461  452797   \n",
       "63    485      1  1804496  521213   \n",
       "64    487      1  1804548  224717   \n",
       "65    488      1  1804565  224716   \n",
       "66    489      1  1806476  422536   \n",
       "67    490      1  1806503  391609   \n",
       "68    491      1  1806503  208722   \n",
       "69    492      1  1806503  224715   \n",
       "70    493      1  1806509  224714   \n",
       "71    494      1  1806512  431082   \n",
       "72    495      1  1806513  224713   \n",
       "73    496      1  1806514  224712   \n",
       "74    497      1  1806515  224711   \n",
       "75    498      1  1815801  617622   \n",
       "76    519      1  2027636   16212   \n",
       "77    568      1  2228828  224869   \n",
       "78    569      1  2228834  570752   \n",
       "79    571      1  2228860   37259   \n",
       "80    573      1  2228866   37261   \n",
       "81    574      1  2228866   37262   \n",
       "82    575      1  2228867   37260   \n",
       "83    576      1  2228867   39783   \n",
       "84    578      1  2228870   39785   \n",
       "85    599      1  2229045   37263   \n",
       "86    600      1  2229045   39784   \n",
       "87    605      1  2229113   37258   \n",
       "88    606      1  2229115  224868   \n",
       "89    607      1  2229118  222819   \n",
       "90    622      1  2229265  213703   \n",
       "91    626      1  2229305  495053   \n",
       "92    634      1  2229401  463409   \n",
       "93    846      1  2406483  554342   \n",
       "94    847      1  2406483  557020   \n",
       "95    850      1  2406526    6771   \n",
       "96    852      1  2406528  197886   \n",
       "97    857      1  2406561  282334   \n",
       "98    862      1  2406580  555297   \n",
       "99    863      1  2406580  296273   \n",
       "\n",
       "                                                  REF        ALT  \\\n",
       "0                                                   C          T   \n",
       "1                                                   C         CG   \n",
       "2                                                   G          T   \n",
       "3                                                   G          A   \n",
       "4                                                   A          T   \n",
       "5                 CGCCCGCCAGGAGAATGTCTTCAAGAAGTTCGACG          C   \n",
       "6                                                   C          T   \n",
       "7                                                   T         TC   \n",
       "8                                                   C          T   \n",
       "9                                                   G          A   \n",
       "10                                                  G          C   \n",
       "11                                                  G          T   \n",
       "12                                                  C         CT   \n",
       "13                                                  G          A   \n",
       "14                                                  G          A   \n",
       "15                                                  A          G   \n",
       "16                                                  C          T   \n",
       "17                                                  A          G   \n",
       "18                                                  C          T   \n",
       "19                                                  G          A   \n",
       "20                            CGGCGCGCCCTGGAGCGGGAGCA          C   \n",
       "21                                                 GA          G   \n",
       "22                                         GTGCTGGCCA          G   \n",
       "23                                                  G          A   \n",
       "24                                                  T          C   \n",
       "25                                                  C         CG   \n",
       "26                                                 CG          C   \n",
       "27                                                  G          C   \n",
       "28                                                  G          A   \n",
       "29                                                  C          T   \n",
       "30                                                  G          C   \n",
       "31                                                  T          A   \n",
       "32                                                 AC          A   \n",
       "33                                     CTGCTGACTCCCGG          C   \n",
       "34                                                 TC          T   \n",
       "35                                  TCCCGGTGCTGCCGCTG          T   \n",
       "36                                                  G     GCTGCC   \n",
       "37                                                 CT          C   \n",
       "38                                           TAGGCAGG          C   \n",
       "39                                                 AA          G   \n",
       "40                                                 TG          T   \n",
       "41                                                 AC          A   \n",
       "42                                                 AG          A   \n",
       "43                                                 CA          C   \n",
       "44                                     GGGGGCAGCCGGGT          G   \n",
       "45                                                 CG          C   \n",
       "46                                     CGGGTGGGGCAGCG          C   \n",
       "47                                                 TG          T   \n",
       "48                                                  G  GGCATTGGC   \n",
       "49                                                  A          T   \n",
       "50                                                 GC          G   \n",
       "51                                                  C          T   \n",
       "52                                                  C          T   \n",
       "53                                                  G          A   \n",
       "54                                                  G          A   \n",
       "55                                                  G          C   \n",
       "56                                                  G          A   \n",
       "57                                                  C          T   \n",
       "58                                                  G          A   \n",
       "59                                                  C          T   \n",
       "60                                               GATC          G   \n",
       "61                                                  C          T   \n",
       "62                                                  C          T   \n",
       "63                                                  T          C   \n",
       "64                                                  T          C   \n",
       "65                                                  A          G   \n",
       "66                                                  T          C   \n",
       "67                                                  A          C   \n",
       "68                                                  A          G   \n",
       "69                                                  A          T   \n",
       "70                                                  T          C   \n",
       "71                                                  C          G   \n",
       "72                                                  C          T   \n",
       "73                                                  A          C   \n",
       "74                                                  T          C   \n",
       "75                                                  C          T   \n",
       "76                                                  A          C   \n",
       "77                                                  T          G   \n",
       "78                                                  A          C   \n",
       "79                                                  C          G   \n",
       "80                                                  G          A   \n",
       "81                                                  G          T   \n",
       "82                                                  G          A   \n",
       "83                                                  G          T   \n",
       "84                                                  C          A   \n",
       "85                                         GTCCGACCGC          G   \n",
       "86                                      GTCCGACCGCTCC          G   \n",
       "87                                                  G          A   \n",
       "88                                                  G          C   \n",
       "89                                                  G          A   \n",
       "90                                                 CT         GG   \n",
       "91                                                  C          A   \n",
       "92  AGCTCAGCGAGCGCAGCGTCCGCGTGTACCACGAGTGCTTCGGCAA...          A   \n",
       "93                                                  C          G   \n",
       "94                                                  C          T   \n",
       "95                                                  G          C   \n",
       "96                                                  G         GC   \n",
       "97                                                  C          A   \n",
       "98                                                 CA          C   \n",
       "99                                                CAG          C   \n",
       "\n",
       "                                                CLNDN  \\\n",
       "0   Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "1   Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "2   Immunodeficiency_38_with_basal_ganglia_calcifi...   \n",
       "3                      Congenital_myasthenic_syndrome   \n",
       "4                      Congenital_myasthenic_syndrome   \n",
       "5                  Myasthenic_syndrome,_congenital,_8   \n",
       "6   Congenital_myasthenic_syndrome|Myasthenic_synd...   \n",
       "7                      Congenital_myasthenic_syndrome   \n",
       "8                                           Not_Given   \n",
       "9                      Congenital_myasthenic_syndrome   \n",
       "10  Congenital_myasthenic_syndrome|Myasthenic_synd...   \n",
       "11  Congenital_myasthenic_syndrome|Myasthenic_synd...   \n",
       "12                                          Not_Given   \n",
       "13                     Congenital_myasthenic_syndrome   \n",
       "14                                Immunodeficiency_16   \n",
       "15  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "16          Ehlers-Danlos_syndrome,_progeroid_type,_2   \n",
       "17  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "18  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "19                                          Not_Given   \n",
       "20  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "21          Ehlers-Danlos_syndrome,_progeroid_type,_2   \n",
       "22          Ehlers-Danlos_syndrome,_progeroid_type,_2   \n",
       "23  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "24  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "25                                          Not_Given   \n",
       "26          Ehlers-Danlos_syndrome,_progeroid_type,_2   \n",
       "27  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "28  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "29  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "30  Spondyloepimetaphyseal_dysplasia_with_joint_la...   \n",
       "31          Ehlers-Danlos_syndrome,_progeroid_type,_2   \n",
       "32                                          Not_Given   \n",
       "33                                          Not_Given   \n",
       "34             Robinow_syndrome,_autosomal_dominant_2   \n",
       "35             Robinow_syndrome,_autosomal_dominant_2   \n",
       "36             Robinow_syndrome,_autosomal_dominant_2   \n",
       "37             Robinow_syndrome,_autosomal_dominant_2   \n",
       "38             Robinow_syndrome,_autosomal_dominant_2   \n",
       "39             Robinow_syndrome,_autosomal_dominant_2   \n",
       "40             Robinow_syndrome,_autosomal_dominant_2   \n",
       "41             Robinow_syndrome,_autosomal_dominant_2   \n",
       "42             Robinow_syndrome,_autosomal_dominant_2   \n",
       "43             Robinow_syndrome,_autosomal_dominant_2   \n",
       "44             Robinow_syndrome,_autosomal_dominant_2   \n",
       "45             Robinow_syndrome,_autosomal_dominant_2   \n",
       "46             Robinow_syndrome,_autosomal_dominant_2   \n",
       "47                            Inborn_genetic_diseases   \n",
       "48                                          Not_Given   \n",
       "49                    Abnormality_of_brain_morphology   \n",
       "50                                Harel-Yoon_syndrome   \n",
       "51                   Harel-Yoon_syndrome|not_provided   \n",
       "52                                Harel-Yoon_syndrome   \n",
       "53                          Spinocerebellar_ataxia_21   \n",
       "54             Spinocerebellar_ataxia_21|not_provided   \n",
       "55                          Spinocerebellar_ataxia_21   \n",
       "56                          Spinocerebellar_ataxia_21   \n",
       "57                          Spinocerebellar_ataxia_21   \n",
       "58                          Spinocerebellar_ataxia_21   \n",
       "59                                          Not_Given   \n",
       "60                            Inborn_genetic_diseases   \n",
       "61  Neurodevelopmental_Disability|hypotonia|Intell...   \n",
       "62                                          Not_Given   \n",
       "63                            Inborn_genetic_diseases   \n",
       "64  Neurodevelopmental_Disability|hypotonia|Intell...   \n",
       "65  Neurodevelopmental_Disability|hypotonia|Strabi...   \n",
       "66                                          Not_Given   \n",
       "67                                          Not_Given   \n",
       "68  Neurodevelopmental_Disability|hypotonia|Cleft_...   \n",
       "69  Neurodevelopmental_Disability|hypotonia|Seizur...   \n",
       "70  Neurodevelopmental_Disability|hypotonia|Strabi...   \n",
       "71          Mental_retardation,_autosomal_dominant_42   \n",
       "72  Neurodevelopmental_Disability|hypotonia|Seizur...   \n",
       "73  Neurodevelopmental_Disability|hypotonia|Strabi...   \n",
       "74  Neurodevelopmental_Disability|hypotonia|Intell...   \n",
       "75          Mental_retardation,_autosomal_dominant_42   \n",
       "76  Generalized_epilepsy_with_febrile_seizures_plu...   \n",
       "77                       Shprintzen-Goldberg_syndrome   \n",
       "78                       Shprintzen-Goldberg_syndrome   \n",
       "79                       Shprintzen-Goldberg_syndrome   \n",
       "80  Inborn_genetic_diseases|Shprintzen-Goldberg_sy...   \n",
       "81  Inborn_genetic_diseases|Shprintzen-Goldberg_sy...   \n",
       "82                       Shprintzen-Goldberg_syndrome   \n",
       "83                       Shprintzen-Goldberg_syndrome   \n",
       "84                       Shprintzen-Goldberg_syndrome   \n",
       "85                       Shprintzen-Goldberg_syndrome   \n",
       "86                       Shprintzen-Goldberg_syndrome   \n",
       "87                       Shprintzen-Goldberg_syndrome   \n",
       "88          Shprintzen-Goldberg_syndrome|not_provided   \n",
       "89                       Shprintzen-Goldberg_syndrome   \n",
       "90                                          Not_Given   \n",
       "91                      Disproportionate_tall_stature   \n",
       "92                       Shprintzen-Goldberg_syndrome   \n",
       "93  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...   \n",
       "94  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...   \n",
       "95                  Peroxisome_biogenesis_disorder_6B   \n",
       "96                                          Not_Given   \n",
       "97                                          Not_Given   \n",
       "98  Peroxisome_biogenesis_disorder_6A|Peroxisome_b...   \n",
       "99  Zellweger_syndrome|Peroxisome_biogenesis_disor...   \n",
       "\n",
       "                          CLNSIG        GENEINFO  IMPLIEDCLASS  \n",
       "0                     Pathogenic      ISG15:9636          1.00  \n",
       "1                     Pathogenic      ISG15:9636          1.00  \n",
       "2                     Pathogenic      ISG15:9636          1.00  \n",
       "3                     Pathogenic     AGRN:375790          1.00  \n",
       "4                     Pathogenic     AGRN:375790          1.00  \n",
       "5                     Pathogenic     AGRN:375790          1.00  \n",
       "6                     Pathogenic     AGRN:375790          1.00  \n",
       "7                     Pathogenic     AGRN:375790          1.00  \n",
       "8              Likely_pathogenic     AGRN:375790          1.00  \n",
       "9                     Pathogenic     AGRN:375790          1.00  \n",
       "10                    Pathogenic     AGRN:375790          1.00  \n",
       "11                    Pathogenic     AGRN:375790          1.00  \n",
       "12             Likely_pathogenic     AGRN:375790          1.00  \n",
       "13                    Pathogenic     AGRN:375790          1.00  \n",
       "14                    Pathogenic    TNFRSF4:7293          1.00  \n",
       "15                    Pathogenic  B3GALT6:126792          1.00  \n",
       "16                    Pathogenic  B3GALT6:126792          1.00  \n",
       "17                    Pathogenic  B3GALT6:126792          1.00  \n",
       "18                    Pathogenic  B3GALT6:126792          1.00  \n",
       "19             Likely_pathogenic  B3GALT6:126792          1.00  \n",
       "20                    Pathogenic  B3GALT6:126792          1.00  \n",
       "21                    Pathogenic  B3GALT6:126792          1.00  \n",
       "22                    Pathogenic  B3GALT6:126792          1.00  \n",
       "23                    Pathogenic  B3GALT6:126792          1.00  \n",
       "24                    Pathogenic  B3GALT6:126792          1.00  \n",
       "25                    Pathogenic  B3GALT6:126792          1.00  \n",
       "26                    Pathogenic  B3GALT6:126792          1.00  \n",
       "27                    Pathogenic  B3GALT6:126792          1.00  \n",
       "28                    Pathogenic  B3GALT6:126792          1.00  \n",
       "29                    Pathogenic  B3GALT6:126792          1.00  \n",
       "30                    Pathogenic  B3GALT6:126792          1.00  \n",
       "31                    Pathogenic  B3GALT6:126792          1.00  \n",
       "32             Likely_pathogenic       DVL1:1855          1.00  \n",
       "33             Likely_pathogenic       DVL1:1855          1.00  \n",
       "34             Likely_pathogenic       DVL1:1855          1.00  \n",
       "35             Likely_pathogenic       DVL1:1855          1.00  \n",
       "36             Likely_pathogenic       DVL1:1855          1.00  \n",
       "37                    Pathogenic       DVL1:1855          1.00  \n",
       "38                    Pathogenic       DVL1:1855          1.00  \n",
       "39                    Pathogenic       DVL1:1855          1.00  \n",
       "40                    Pathogenic       DVL1:1855          1.00  \n",
       "41                    Pathogenic       DVL1:1855          1.00  \n",
       "42                    Pathogenic       DVL1:1855          1.00  \n",
       "43                    Pathogenic       DVL1:1855          1.00  \n",
       "44                    Pathogenic       DVL1:1855          1.00  \n",
       "45                    Pathogenic       DVL1:1855          1.00  \n",
       "46             Likely_pathogenic       DVL1:1855          1.00  \n",
       "47                    Pathogenic       DVL1:1855          1.00  \n",
       "48             Likely_pathogenic       DVL1:1855          1.00  \n",
       "49             Likely_pathogenic     MXRA8:54587          1.00  \n",
       "50             Likely_pathogenic    ATAD3A:55210          1.00  \n",
       "51             Likely_pathogenic    ATAD3A:55210          1.00  \n",
       "52                    Pathogenic    ATAD3A:55210          1.00  \n",
       "53                    Pathogenic  TMEM240:339453          1.00  \n",
       "54  Pathogenic/Likely_pathogenic  TMEM240:339453          1.00  \n",
       "55                    Pathogenic  TMEM240:339453          1.00  \n",
       "56                    Pathogenic  TMEM240:339453          1.00  \n",
       "57                    Pathogenic  TMEM240:339453          1.00  \n",
       "58                    Pathogenic  TMEM240:339453          1.00  \n",
       "59             Likely_pathogenic  TMEM240:339453          1.00  \n",
       "60                    Pathogenic       GNB1:2782          1.00  \n",
       "61  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "62             Likely_pathogenic       GNB1:2782          1.00  \n",
       "63                    Pathogenic       GNB1:2782          1.00  \n",
       "64  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "65  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "66             Likely_pathogenic       GNB1:2782          1.00  \n",
       "67                    Pathogenic       GNB1:2782          1.00  \n",
       "68                    Pathogenic       GNB1:2782          1.00  \n",
       "69  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "70                    Pathogenic       GNB1:2782          1.00  \n",
       "71             Likely_pathogenic       GNB1:2782          1.00  \n",
       "72             Likely_pathogenic       GNB1:2782          1.00  \n",
       "73  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "74  Pathogenic/Likely_pathogenic       GNB1:2782          1.00  \n",
       "75             Likely_pathogenic       GNB1:2782          1.00  \n",
       "76                   risk_factor      GABRD:2563          1.00  \n",
       "77                    Pathogenic        SKI:6497          1.00  \n",
       "78             Likely_pathogenic        SKI:6497          1.00  \n",
       "79                    Pathogenic        SKI:6497          1.00  \n",
       "80  Pathogenic/Likely_pathogenic        SKI:6497          1.00  \n",
       "81                    Pathogenic        SKI:6497          1.00  \n",
       "82                    Pathogenic        SKI:6497          1.00  \n",
       "83                    Pathogenic        SKI:6497          1.00  \n",
       "84                    Pathogenic        SKI:6497          1.00  \n",
       "85                    Pathogenic        SKI:6497          1.00  \n",
       "86                    Pathogenic        SKI:6497          1.00  \n",
       "87                    Pathogenic        SKI:6497          1.00  \n",
       "88  Pathogenic/Likely_pathogenic        SKI:6497          1.00  \n",
       "89             Likely_pathogenic        SKI:6497          1.00  \n",
       "90             Likely_pathogenic        SKI:6497          1.00  \n",
       "91             Likely_pathogenic        SKI:6497          1.00  \n",
       "92                    Pathogenic        SKI:6497          1.00  \n",
       "93             Likely_pathogenic      PEX10:5192          1.00  \n",
       "94             Likely_pathogenic      PEX10:5192          1.00  \n",
       "95                    Pathogenic      PEX10:5192          1.00  \n",
       "96                    Pathogenic      PEX10:5192          1.00  \n",
       "97                    Pathogenic      PEX10:5192          1.00  \n",
       "98             Likely_pathogenic      PEX10:5192          1.00  \n",
       "99                    Pathogenic      PEX10:5192          1.00  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ==============================================================================================================\n",
    "# STEP 3c: PERFORM INITIAL/BASIC EDA\n",
    "# ==============================================================================================================\n",
    "\n",
    "# Determine counts of potentially dangerous/harmful and non-dangerous/non-harmful variants in dataframe\n",
    "print(\"Number of Potentially Dangerous/Harmful Variants (IMPLIEDCLASS = 1):\")\n",
    "print(clinical_filtered_df[\"IMPLIEDCLASS\"].sum(), end = \"\\n\\n\")\n",
    "\n",
    "# Determine percentage of potentially dangerous/harmful variants out of all variants in dataframe\n",
    "print(\"Percentage of Potentially Dangerous/Harmful Variants out of Total Variants:\")\n",
    "print(round(clinical_filtered_df[\"IMPLIEDCLASS\"].sum() / len(clinical_filtered_df), 2) * 100)\n",
    "\n",
    "# Filter dataframe to just potentially dangerous/harmful variants (IMPLIEDCLASS = 1)\n",
    "clinical_harmful_df = clinical_filtered_df[clinical_filtered_df[\"IMPLIEDCLASS\"] == 1]\n",
    "\n",
    "# Create final dataframe, limited to only first 100 potentially dangerous/harmful variants in dataset\n",
    "clinical_final_df = clinical_harmful_df[:100].reset_index()\n",
    "\n",
    "# Display distribution (counts) of variants by clinical disease name to see unique/highest-density CLNDN values\n",
    "# clinical_final_df[\"CLNDN\"].value_counts()\n",
    "\n",
    "# Display final dataframe\n",
    "clinical_final_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**ASSUMPTIONS:**\n",
    "- The first 27 rows of the vcf file/dataset do not need to be included in the final dataframe because they simply contain meta-information. I'm assuming it's okay to not even read them into the original dataframe.\n",
    "- The POS column from the original vcf file/dataset contains a mix of Integer and String values. Based on the dataset documentation, this column _should_ be in **Integer** form only. Some of the values contain \"O\" characters in addition to numbers/digits, making them Strings. I'm assuming these \"O\" values were unintentional and should be replaced with 0s (which provides an opportunity for data-cleaning).\n",
    "- After performing the parsing step on the INFO column (step 2b), I'm assuming that no additional parsing of any of the INFO sub-components/sub-fields (e.g., AF_ESP, AF_EXAC, etc.) is necessary for the ask from the boss.\n",
    "- The \"SSR\" sub-component/sub-field within the INFO column is actually non-existent in the data (though it's mentioned in the meta-information data for the vcf file). I'm assuming this to be the case given the output of the INFO column parsing step (resulting from json.normalize in step 2b) - which should capture data for each sub-component/sub-field of INFO _wherever data actually exists_. To be extra sure, I also imported the vcf file data into an Excel workbook and performed a Find search for \"SSR\" in the INFO column and found no hits.\n",
    "- The exact CLNSIG descriptions I categorized as potentially dangerous/harmful are shown in step 3b. I chose these particular descriptions because they were either defined as pathogenic, likely pathogenic, or as a risk factor, which implies that there's a high probability each of these variants is harmful. I chose not to include all other CLNSIG descriptions in my potentially dangerous/harmful categorization (including descriptions such as benign, likely benign, inconclusive, conflicting, etc.) because I assumed they all had a low (or no) probability of actually being harmful.\n",
    "- I'm assuming that _first 100 harmful mutations_ (as described in the \"Business Case\" above) literally means the first 100 harmful mutations that appear in the dataset, starting at the top (index/row 0). I am _not_ assuming that statement to mean the top 100 most common harmful mutations in the dataset (which would be based on counts of occurrence).\n",
    "\n",
    "Assumed Mappings of Required Fields to Dataframe Column Names (based on documentation) - keep these columns:\n",
    "- 1) Gene name -> **GENEINFO** column\n",
    "- 2) Mutation ID number -> **ID** column\n",
    "- 3) Mutation Position (chromosome & position) -> **CHROM** and **POS** columns\n",
    "- 4) Mutation value (reference & alternate bases) -> **REF** and **ALT** columns\n",
    "- 5) Clinical significance (CLNSIG) -> **CLNSIG** column\n",
    "- 6) Disease that is implicated -> **CLNDN** column"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### What would you present to your boss?\n",
    "\n",
    "The final dataframe I would share with my boss is displayed under step 3c above. This dataframe output shows **the first 100 potentially dangerous/harmful variants in the dataset.**\n",
    "\n",
    "I would also let my boss know that after performing some initial/basic EDA on the dataset and grouping variants by their clinical significance (CLNSIG), I'd conclude that **approximately 19% of all variants in the file - or roughly 19,572 records out of 102,321 records in total - represent potentially dangerous/harmful variants.**\n",
    "\n",
    "To support these numbers, I would share with my boss how I classified potentially dangerous/harmful variants and potentially non-dangerous/non-harmful variants. For transparency, I would briefly summarize **all of the assumptions I listed above** but would specifically call out bullet point #5, mentioning that my classification of variants was based on the CLNSIG descriptions in the file and was relatively intuitive (i.e., I treated \"pathogenic\"/\"likely-pathogenic\" variants as potentially dangerous/harmful and treated _most_ others as non-dangerous/non-harmful). Of course, this classification scheme also makes the baseline assumption that the CLNSIG data is credible, so I would qualify everything I share with a statement that I could certainly learn more about this data and would be eager to spend more time getting comfortable with the documentation before we make any recommendations or decisions. (For instance, I'd probably say something along these lines: \"I know you pulled this file off the Internet and had me dive into it right away to take a look - and that's awesome - but I'm also still relatively new to everything that's in here and what it all means ... so this is _truly_ just a first pass at some initial insights!\")\n",
    "\n",
    "A cursory glance at the final dataframe reveals that a variety of diseases (CLNDN descriptions) are implicated in the first 100 records. For instance, I see a handful of hits for \"Congenital_myasthenic_syndrome\", \"Robinow_syndrome\", \"Neurodevelopmental_Disability\", and quite a few others. Based on this quick review, I'd mention to my boss that **a deeper-dive into the CLNDN data - as well as several of the other data columns - would be helpful in order to better understand the exact gene and mutation combinations that produce (or lead to) dangerous/harmful conditions.** As a next step, we could take a closer look at how much overlap there might be - if any - between some of these diseases and the gene/mutation combinations listed in the file (e.g., do some gene/mutation combinations produce a wider array of dangerous/harmful results - manifested in multiple different types of diseases - than others). We could also try to learn more about the CLNSIG descriptions and confirm their accuracy (or the overall confidence level put into each category), and see if there are certain descriptions/categories that are more severe - or that have a higher degree of confidence around pathogenicity - than others."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
